### ISSUE TITLE
lists not working with Custom objects

### ISSUE THREAD

**Issue by DiellzaElshani** [BHoM/RDF_Prototypes#29]
#### Description:

The current TTLgraph component can handle rdf lists (as rdf seq), however it does not recognize lists that are inputted in a BHoM Costume Object. Regardless if is a list of literals or objects, we get a Basse64JsonSerialized dataype as output. See figure: 

![image](https://user-images.githubusercontent.com/50298535/175576128-e4f73ea7-6b98-4c82-8d8d-dabe0a136367.png)

#### Example object:

To test the behavior, here is a costume object which takes as input a Type: Casette; and 3 Points - ControlPoints as a list. 

![sample object](https://user-images.githubusercontent.com/50298535/175576919-84977c08-5b07-4326-a71e-db066257e401.png)

#### Current output:

```
#################################################################
#    Individuals
#################################################################


### http://www.uni-stuttgart.de/9ff6861d-4626-4c90-9bbe-391644dbb79d
<http://www.uni-stuttgart.de/9ff6861d-4626-4c90-9bbe-391644dbb79d> rdf:type owl:NamedIndividual ,
		:BH.Engine.RDF.Types.CustomObjectType.Casette ;
		:BH.Engine.RDF.Types.CustomObjectType.Casette.ControlPoints "W3siJHR5cGUiOiJCSC5vTS5HZW9tZXRyeS5Qb2ludCwgR2VvbWV0cnlfb00iLCJYIjowLjAsIlkiOjAuMCwiWiI6MC4wfSx7IiR0eXBlIjoiQkgub00uR2VvbWV0cnkuUG9pbnQsIEdlb21ldHJ5X29NIiwiWCI6MC4wLCJZIjowLjAsIloiOjAuMH0seyIkdHlwZSI6IkJILm9NLkdlb21ldHJ5LlBvaW50LCBHZW9tZXRyeV9vTSIsIlgiOjAuMCwiWSI6MC4wLCJaIjowLjB9XQ=="^^:BH.oM.RDF.Base64JsonSerialized.
```

The correct ttl output should be: 

```


#################################################################
#    Individuals
#################################################################


### http://www.uni-stuttgart.de/6c96428b-4696-457a-a42d-6a2bca5fdc33
<http://www.uni-stuttgart.de/6c96428b-4696-457a-a42d-6a2bca5fdc33> rdf:type owl:NamedIndividual ,
		:BH.Engine.RDF.Types.CustomObjectType.Casette ;
		:BH.Engine.RDF.Types.CustomObjectType.Casette.ControlPoints rdf:seq ;
		rdf:_0 <http://www.uni-stuttgart.de/206839> ;
		rdf:_1 <http://www.uni-stuttgart.de/1363879927> ;
		rdf:_2 <http://www.uni-stuttgart.de/1073948663> .


### http://www.uni-stuttgart.de/206839
<http://www.uni-stuttgart.de/206839> rdf:type owl:NamedIndividual ,
		:BH.oM.Geometry.Point ;
		:BH.oM.Geometry.Point.X "0"^^xsd:double;
		:BH.oM.Geometry.Point.Y "0"^^xsd:double;
		:BH.oM.Geometry.Point.Z "0"^^xsd:double.


### http://www.uni-stuttgart.de/1363879927
<http://www.uni-stuttgart.de/1363879927> rdf:type owl:NamedIndividual ,
		:BH.oM.Geometry.Point ;
		:BH.oM.Geometry.Point.X "3"^^xsd:double;
		:BH.oM.Geometry.Point.Y "3"^^xsd:double;
		:BH.oM.Geometry.Point.Z "3"^^xsd:double.


### http://www.uni-stuttgart.de/1073948663
<http://www.uni-stuttgart.de/1073948663> rdf:type owl:NamedIndividual ,
		:BH.oM.Geometry.Point ;
		:BH.oM.Geometry.Point.X "2"^^xsd:double;
		:BH.oM.Geometry.Point.Y "2"^^xsd:double;
		:BH.oM.Geometry.Point.Z "2"^^xsd:double.

```

(In the next comment I will add the full ttl text).

**Comment by DiellzaElshani** [comment]
```
@prefix : <www.bhom.org/> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix xml: <http://www.w3.org/XML/1998/namespace> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix dc: <http://purl.org/dc/elements/1.1/> .
@base   <www.bhom.org> .

<www.bhom.org> rdf:type owl:Ontology;
                          dc:title "buidling ontology"@en;
                          dc:description " bhom ontology"@en.


#################################################################
#    Annotation Properties
#################################################################

###  http://purl.org/dc/elements/1.1/#description
<http://purl.org/dc/elements/1.1/#description> rdf:type owl:AnnotationProperty .
###  http://purl.org/dc/elements/1.1/#title
<http://purl.org/dc/elements/1.1/#title> rdf:type owl:AnnotationProperty .


#################################################################
#    Datatypes
#################################################################

### https://github.com/BHoM/RDF_Prototypes/commit/ff8ccb68dbba5aeadb4a9a284f141eb1515e169a
:BH.oM.RDF.Base64JsonSerialized rdf:type rdfs:Datatype ;
rdfs:label "Base64JsonSerialized"@en ;
owl:equivalentClass xsd:string .


#################################################################
#    Classes
#################################################################

### http://customizemefrom-ontologysettings.tboxsettings.customobjecttypesbaseaddress/Casette
:BH.Engine.RDF.Types.CustomObjectType.Casette rdf:type owl:Class;
		rdfs:subClassOf :BH.oM.Base.CustomObject;
		rdfs:label "Casette"@en .

### https://github.com/BHoM/BHoM/blob/main/BHoM/Interface/IBHoMObject.cs
:BH.oM.Base.IBHoMObject rdf:type owl:Class;
		rdfs:subClassOf :BH.oM.Base.IObject;
		rdfs:label "IBHoMObject"@en .

### https://github.com/BHoM/BHoM/blob/main/BHoM/Interface/IObject.cs
:BH.oM.Base.IObject rdf:type owl:Class;
		rdfs:label "IObject"@en .

### https://github.com/BHoM/BHoM/blob/main/BHoM/CustomObject.cs
:BH.oM.Base.CustomObject rdf:type owl:Class;
		rdfs:subClassOf :BH.oM.Base.BHoMObject;
		rdfs:label "CustomObject"@en .

### https://github.com/BHoM/BHoM/blob/main/BHoM/BHoMObject.cs
:BH.oM.Base.BHoMObject rdf:type owl:Class;
		rdfs:subClassOf :BH.oM.Base.IBHoMObject;
		rdfs:label "BHoMObject"@en .


#################################################################
#    Object Properties
#################################################################




#################################################################
#    Data properties
#################################################################


### http://customizemefrom-ontologysettings.tboxsettings.customobjecttypesbaseaddress/Casette#ControlPoints
:BH.Engine.RDF.Types.CustomObjectType.Casette.ControlPoints rdf:type owl:DatatypeProperty ;
rdfs:domain :BH.Engine.RDF.Types.CustomObjectType.Casette ;
rdfs:range :BH.oM.RDF.Base64JsonSerialized ;
rdfs:label "ControlPoints"@en .


#################################################################
#    Individuals
#################################################################


### http://www.uni-stuttgart.de/6c96428b-4696-457a-a42d-6a2bca5fdc33
<http://www.uni-stuttgart.de/6c96428b-4696-457a-a42d-6a2bca5fdc33> rdf:type owl:NamedIndividual ,
		:BH.Engine.RDF.Types.CustomObjectType.Casette ;
		:BH.Engine.RDF.Types.CustomObjectType.Casette.ControlPoints rdf:seq ;
		rdf:_0 <http://www.uni-stuttgart.de/206839> ;
		rdf:_1 <http://www.uni-stuttgart.de/1363879927> ;
		rdf:_2 <http://www.uni-stuttgart.de/1073948663> .


### http://www.uni-stuttgart.de/206839
<http://www.uni-stuttgart.de/206839> rdf:type owl:NamedIndividual ,
		:BH.oM.Geometry.Point ;
		:BH.oM.Geometry.Point.X "0"^^xsd:double;
		:BH.oM.Geometry.Point.Y "0"^^xsd:double;
		:BH.oM.Geometry.Point.Z "0"^^xsd:double.


### http://www.uni-stuttgart.de/1363879927
<http://www.uni-stuttgart.de/1363879927> rdf:type owl:NamedIndividual ,
		:BH.oM.Geometry.Point ;
		:BH.oM.Geometry.Point.X "3"^^xsd:double;
		:BH.oM.Geometry.Point.Y "3"^^xsd:double;
		:BH.oM.Geometry.Point.Z "3"^^xsd:double.


### http://www.uni-stuttgart.de/1073948663
<http://www.uni-stuttgart.de/1073948663> rdf:type owl:NamedIndividual ,
		:BH.oM.Geometry.Point ;
		:BH.oM.Geometry.Point.X "2"^^xsd:double;
		:BH.oM.Geometry.Point.Y "2"^^xsd:double;
		:BH.oM.Geometry.Point.Z "2"^^xsd:double.




```

**Comment by alelom** [comment]
Closed with 22f24387a002c8cf9a551bebd45bb7a85a2976c8 and previous commits.
